### DISCUSSION TITLE
AWS S3 bucket with temporary credentials

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#3861] [Feature Requests]
If an account is using AWS temporarily credentials, it requires to pass the following values:

- `AWS_SECRET_ACCESS_KEY` 
- `AWS_ACCESS_KEY_ID` 
- `AWS_SESSION_TOKEN`

The `AWS_SECRET_ACCESS_KEY` is equal to `STORAGE_AMAZON_KEY` and the `AWS_ACCESS_KEY_ID` is equal to `STORAGE_AMAZON_SECRET`, but the `AWS_SESSION_TOKEN` is not available and cannot be set.

As a workaround on my Directus `v9.0.0-rc.34` I added the variables:

- `AWS_SECRET_ACCESS_KEY` 
- `AWS_ACCESS_KEY_ID` 
- `AWS_SESSION_TOKEN`

It works, but it is using the AWS SDK directly instead of the Directus setup.

**Comment by benhaynes** [comment]
@rijkvanzanten @philleepflorence — Docs again... 😄

**Reply by philleepflorence** [reply]
@juancarlosjr97 - would you have a link to a reference for this, so I have a context when adding to the docs?

**Reply by juancarlosjr97** [reply]
Sure, my pleasure.

- Link to the AWS temporary credentials: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html
- Link to the credentials using the AWS SDK JS: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EnvironmentCredentials.html

I hope that is useful @philleepflorence. Let me know if you need anything else.

**Reply by philleepflorence** [reply]
I appreciate the references, they will definitely come in handy...

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
